src/main/java/org/codehaus/groovy/vmplugin/v7/IndyInterface.java

```diff
@@ -258,13 +261,18 @@ public static Object fromCache(MutableCallSite callSite, Class<?> sender, String
         MethodHandleWrapper mhw =
                 doWithCallSite(
                         callSite, arguments,
-                        (cs, receiver) ->
-                                cs.getAndPut(
+                        (CacheableCallSite ccs, Object receiver) ->
+                                ccs.getAndPut(
                                         receiver.getClass().getName(),
-                                        c -> fallback(callSite, sender, methodName, callID, safeNavigation, thisCall, spreadCall, dummyReceiver, arguments)
+                                        c -> Optional.of(fallback(callSite, sender, methodName, callID, safeNavigation, thisCall, spreadCall, dummyReceiver, arguments))
```

Comment:
`Optional.of` will create a new object, which may impact the performance a bit.
Why not just use `... ? ... : ...`?

Reply:
How much performance are we talking about?  This form allows expression lambda.  Otherwise it would need to be restructured to a statement lambda.